sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedDeltaLengthByteArrayReader.java

```diff
@@ -66,20 +57,18 @@ public void readBinary(int total, WritableColumnVector c, int rowId) {
     }
     ByteBuffer buffer;
     ByteBufferOutputWriter outputWriter;
-    if (memoryMode == MemoryMode.OFF_HEAP) {
-      outputWriter = ByteBufferOutputWriter::copyWriteByteBuffer;
-    } else {
-      outputWriter = ByteBufferOutputWriter::writeArrayByteBuffer;
-    }
+    outputWriter = ByteBufferOutputWriter::writeArrayByteBuffer;
     int length;
     for (int i = 0; i < total; i++) {
       length = lengthsVector.getInt(rowId + i);
-      try {
-        buffer = in.slice(length);
-      } catch (EOFException e) {
-        throw new ParquetDecodingException("Failed to read " + length + " bytes");
+      if (length > 0) {
+        try {
+          buffer = in.slice(length);
+        } catch (EOFException e) {
+          throw new ParquetDecodingException("Failed to read " + length + " bytes");
+        }
+        outputWriter.write(c, rowId + i, buffer, length);
       }
-      outputWriter.write(c, rowId + i, buffer, length);
       currentRow++;
```

Comment:
should we change line72 to `currentRow += currentRow + total` and move it out of the looping?